packages/client/hmi-client/src/composables/useCharts.ts

```diff
@@ -468,7 +468,20 @@ export function useCharts(
 					statLayerVariables.push(`${chartData.value?.pyciemssMap[v]}_mean:pre`);
 				}
 			});
-			colorScheme.push(...CATEGORICAL_SCHEME);
+			if (!setting.variableColors || setting.variableColors === undefined) {
+				setting.variableColors = {};
```

Comment:
Hm.. I think we should avoid modifying setting object here as it makes things more difficult to track what's happening in where. We should only read from setting in this file and update should be handled in `useChartSetting.ts`

Reply:
See https://github.com/DARPA-ASKEM/terarium/pull/6488/files#r1946744670